v2/datastream-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/SpannerTransactionWriterDoFn.java

```diff
@@ -177,6 +211,8 @@ public void processElement(ProcessContext c) {
       JsonNode changeEvent = mapper.readTree(msg.getPayload());
 
       JsonNode retryCount = changeEvent.get("_metadata_retry_count");
+      String tableName = changeEvent.get(EVENT_TABLE_NAME_KEY).asText();
+      Map<String, Object> sourceRecord = convertJsonNodeToMap(changeEvent);
```

Comment:
Can we write this piece in a more modular way? Feels like we are overloading `SpannerTransactionWtiterDoFn`. Think about creating an independent PTransform which handles custom transformation as a problem. 

```
If customTransform
     apply PTransform
else
     None
```

You have added a bunch of conditionals throughout this class currently which will make it harder to test.

Reply:
The more I read the latter half of the class the more I am convinced this needs modularity..  Please refactor!

Reply:
Done. Added transformation as a separate step